src/huggingface_hub/repository.py

```diff
@@ -543,14 +556,21 @@ def clone_from(self, repo_url: str, use_auth_token: Union[bool, str, None] = Non
                 repo_url = repo_url.replace("https://", f"https://user:{token}@")
 
                 if namespace == user or namespace in valid_organisations:
-                    api.create_repo(
-                        repo_id,
-                        token=token,
-                        repo_type=self.repo_type,
-                        organization=namespace,
-                        exist_ok=True,
-                        private=self.private,
-                    )
+                    try:
+                        _ = api.model_info(f"{namespace}/{repo_id}")
+                    except HTTPError:
+                        logger.warning(
+                            f"Repo {namespace}/{repo_id} does not exist, creating a new repo"
+                        )
+                        api.create_repo(
+                            repo_id,
+                            token=token,
+                            repo_type=self.repo_type,
+                            organization=namespace,
+                            exist_ok=True,
+                            private=self.private,
+                            space_sdk=space_sdk,
+                        )
```

Comment:
Not super enthusiastic about that part with the double API call but I understand it's needed for spaces we want to clone when no `space_sdk` is passed.